SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialDispatcher.h

```diff
@@ -45,6 +71,11 @@ class SPATIALGDK_API USpatialDispatcher : public UObject
 	UPROPERTY()
 	USpatialStaticComponentView* StaticComponentView;
 
-	UPROPERTY()
-	TMap<uint32, UOpCallbackTemplate*> UserOpCallbacks; // key has to be explicit primitive or UHT complains
+	// This index is incremented and returned every time an AddOpCallback function is called.
+	// CallbackIds enable you to deregister callbacks using the RemoveOpCallback function. 
+	// RunCallbacks is called by the SpatialDispatcher and executes all user registered 
+	// callbacks for the matching component ID and network operation type.
+	FCallbackId NextCallbackId;
+	TMap<Worker_ComponentId, TMap<Worker_OpType, TArray<UserOpCallbackData>>> ComponentOpTypeToCallbackMap;
```

Comment:
What are the benefits of a multi-level map? Wouldn't it be simpler and more cache-efficient to flatten it?

Reply:
Discussed offline, combining componentId and optype to a combined key would be more efficient in this case. However if we did something like enabling users to deleting all callbacks for a particular component ID, the nested map would work better. Since the expected data structure size is small at the moment, and I suspect this may change to encompass things like entityid soon, we agreed not to make this change in the short term.